src/Integration.Vsix.UnitTests/CFamily/CLangAnalyzerTests.cs

```diff
@@ -348,31 +370,42 @@ private static Mock<IRequestFactoryAggregate> CreateRequestFactory(string filePa
             return new TestableCLangAnalyzer(telemetryManager, settings, statusNotifier, logger, issueConverter, requestFactory);
         }
 
+        private void ExecuteAndWaitForCompletion(TestableCLangAnalyzer testSubject, string path, string charset, AnalysisLanguage[] detectedLanguages, IIssueConsumer consumer, CFamilyAnalyzerOptions analysisOptions, CancellationToken cancellationToken)
+        {
+            // The analyzer spawns the analysis on a background thread, so we need to do some extra work
+            // if we want to wait until the work on the background thread has finished.
+
+            var subProcess = new SubProcessSimulator();
+            testSubject.SetCallSubProcessBehaviour(subProcess.CallSubProcess);
+
+            testSubject.ExecuteAnalysis(path, charset, detectedLanguages, consumer, analysisOptions, cancellationToken);
+            subProcess.WaitUntilSubProcessCalledByAnalyzer();
+            subProcess.SignalNoMoreIssues();
+        }
+
         private class TestableCLangAnalyzer : CLangAnalyzer
         {
             public delegate void HandleCallSubProcess(Action<Message> handleMessage, IRequest request, 
                 ISonarLintSettings settings, ILogger logger, CancellationToken cancellationToken);
 
             private HandleCallSubProcess onCallSubProcess;
+            private readonly AutoResetEvent factoryCallCompleted = new AutoResetEvent(false);
+
             public void SetCallSubProcessBehaviour(HandleCallSubProcess onCallSubProcess)
                 => this.onCallSubProcess = onCallSubProcess;
 
-            public int TriggerAnalysisCallCount { get; private set; }
+            public int SubProcessExecutedCount { get; private set; }
 
             public TestableCLangAnalyzer(ITelemetryManager telemetryManager, ISonarLintSettings settings, 
                 IAnalysisStatusNotifier analysisStatusNotifier, ILogger logger, 
                 ICFamilyIssueToAnalysisIssueConverter cFamilyIssueConverter, IRequestFactoryAggregate requestFactory)
                 : base(telemetryManager, settings, analysisStatusNotifier, cFamilyIssueConverter, requestFactory, logger)
             {}
 
-            protected override void TriggerAnalysis(IRequest request, IIssueConsumer consumer, IAnalysisStatusNotifier analysisStatusNotifier, CancellationToken cancellationToken)
-            {
-                TriggerAnalysisCallCount++;
-            }
-
```

Comment:
Previously this method did nothing except count the call i.e. the real code wasn't called.

Now more of the "real" code is executed by the tests.